Tests/Unit/NetworkServer/CupsUpdateInfoResponseHeaderTests.cs

```diff
@@ -0,0 +1,140 @@
+// Copyright (c) Microsoft. All rights reserved.
+// Licensed under the MIT license. See LICENSE file in the project root for full license information.
+
+namespace LoRaWan.Tests.Unit.NetworkServer
+{
+    using System;
+    using System.Buffers;
+    using System.Collections.Immutable;
+    using System.Linq;
+    using LoRaWan.NetworkServer.BasicsStation;
+    using Xunit;
+
+    public class CupsUpdateInfoResponseHeaderTests
+    {
+        private readonly string updateUriString = "https://localhost:1234";
+
+        [Fact]
+        public void Serialize_WithNoUpdates()
+        {
+            var updateResponseHeader = new CupsUpdateInfoResponseHeader
+            {
+                SignatureKeyCrc = 0,
+                UpdateDataLength = 0,
+            };
+
+            using var memoryPool = MemoryPool<byte>.Shared.Rent(2048);
+
+            // Act
+            var serializedResponse = updateResponseHeader.Serialize(memoryPool.Memory.Span);
+
+            // Assert
+            Assert.True(serializedResponse.ToArray().All(b => b == 0));
```

Comment:
I would assert here in all other tests whether the response was the expected size:

```suggestion
            Assert.Equal(14, serializedResponse.Length);
            Assert.True(serializedResponse.ToArray().All(b => b == 0));
```

Reply:
I would initialise the rented memory with some non-zero/unexpected filler:

```c#
memoryPool.Memory.Span.Fill(0xbd);
```

Suppose also we have a method on `ByteSpanReader` that reads the rest into a returned array:

```c#
public byte[] ReadAll() => Read(this.span.Length);
```

Then you can assert that `Serialize` didn't write _too much_ and remaining bytes in the rented memory are still intact:

```c#
var reader = serializedResponse.GetReader();
Assert.True(reader.ReadAll().All(b => b == 0));
Assert.True(memoryPool.Memory.Span[serializedResponse.Length..].ToArray().All(b => b == 0xbd));
```

I would do this for all other tests too.